server/apps/immich/src/api-v1/share/share.controller.ts

```diff
@@ -0,0 +1,26 @@
+import { Controller, Get, Param, Delete } from '@nestjs/common';
+import { ShareService } from './share.service';
+import { Authenticated } from '../../decorators/authenticated.decorator';
+import { AuthUserDto, GetAuthUser } from '../../decorators/auth-user.decorator';
+import { ApiTags } from '@nestjs/swagger';
+
+@ApiTags('share')
+@Controller('share')
+export class ShareController {
+  constructor(private readonly shareService: ShareService) {}
+  @Authenticated()
+  @Get()
+  getAllSharedLinks(@GetAuthUser() authUser: AuthUserDto) {
+    return this.shareService.findAll(authUser);
+  }
+
+  @Get(':id')
+  getSharedLink(@Param('id') id: string) {
+    return this.shareService.findOne(id);
+  }
+
+  @Delete(':id')
+  removeSharedLink(@Param('id') id: string) {
+    return this.shareService.remove(id);
```

Comment:
These should require authentication. Also, the removeSharedLink should have additional authz that says only the shared link owner can remove it.

Reply:
Ah I had it at one time and remove it for testing, good catch.